### PR TITLE
Fix pre-commit formatting not running on staged files

### PR DESCRIPTION
**Pull request by axbarsan**
- Remove the `pretty-quick` dependency
- Use `lint-staged` to only run tasks on staged files before commiting

Now both linting and formatting are running just on staged files, which increases development speed.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -141,7 +141,14 @@
...
     "hooks": {
-      "pre-commit": "yarn lint:fix && yarn format:staged"
+      "pre-commit": "yarn precommit"
     }
+  },
+  "lint-staged": {
+    "*.js": [
+      "yarn lint:fix",
+      "yarn prettier",
+      "git add"
```
Comment (ubergesundheit) [inline package.json]: Does this automatically run `git add`?

**Comment by ubergesundheit** [comment]
Could you also upgrade husky in this PR to [4.0.10](https://www.npmjs.com/package/husky/v/4.0.10)?

File: package.json (hunk 1 above)
Comment (axbarsan) [reply package.json]: Yes, on the changed files (which are the staged files, anyway), so nothing breaks

Comment (ubergesundheit) [reply package.json]: `changed` or `staged` files? I very much only use `git add -p` for adding only the changes I really want into the staging area. As I understand it, unstaged changes would be added to the staging area with this change?

Comment (axbarsan) [reply package.json]: I apologize for the confusion. I actually meant that it runs on staged files, and after making fixes, it adds the files again, to be able to commit the new changes.

If ran without `git add` as a final task, the changes from the tasks would not be included in the commit.

Comment (ubergesundheit) [reply package.json]: yeah nice :+1:

**Pull request merged**